### PR TITLE
[parametric] Add initial span events system-tests

### PR DESCRIPTION
**Pull request by zacharycmontoya**
## Motivation

We are adding the capability for tracing libraries to implement the following OTel Span API's:
- `AddEvent`
- `RecordException` (a variant of AddEvent)

This PR add tests to set the baseline behavior for the initial set of requirements and runs the test against .NET and Python language implementations (these were tested locally with a compliant implementation).

## Changes

Adds the following parametric tests:
- `Test_Otel_Span_Methods::test_otel_add_event_meta_serialization`
- `Test_Otel_Span_Methods::test_otel_record_exception_does_not_set_error`
- `Test_Otel_Span_Methods::test_otel_record_exception_meta_serialization`

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: tests/parametric/test_otel_span_methods.py [hunk 1]
```diff
@@ -755,6 +756,121 @@ def test_otel_span_extended_reserved_attributes_overrides_analytics_event(
...
+        event3 = events[2]
+        assert event3.get("name") == "exception"
+        assert event3["attributes"].get("exception.message") == "message override"
+        assert event3.get("time_unix_nano") > event2.get("time_unix_nano")
+
+        assert root_span["error"] == 1
+        error_message = root_span["meta"].get("error.message") or root_span["meta"].get("error.msg")
+        assert error_message == "message override"
+        assert "error.type" not in root_span["meta"]
+        assert "error.stack" not in root_span["meta"]
```
Comment (mabdinur) [inline tests/parametric/test_otel_span_methods.py:872]: To maintain compatibility with error tracking should we always attempt set these span tags in the records exception API? 

```suggestion
        assert "error.type" in root_span["meta"]
        assert "error.stack" in root_span["meta"]
```

Comment (mabdinur) [reply tests/parametric/test_otel_span_methods.py:872]: In python:

root_span["meta"]["error.type"] == 'Exception'
root_span["meta"]["error.stack"] == 'Exception: woof3', this stacktrace captures the initial error.message 🤔

Comment (zacharycmontoya) [reply tests/parametric/test_otel_span_methods.py:872]: Yes I think we should try to have `error.type` and `error.stack` be set so your suggestion makes sense. I'm not sure if all of the implementing languages will be able to pass this, so I could see us removing these lines if that turns out to be the case

**Comment by zacharycmontoya** [comment]
Perhaps we need to re-run the failing parametric tests, but the only failures are Errors not Failures.

**Review by cbeauchesne — APPROVED** [review]

File: tests/parametric/test_otel_span_methods.py [hunk 2]
```diff
@@ -789,6 +790,128 @@ def test_otel_span_extended_reserved_attributes_overrides_analytics_event(
...
+        root_span = get_span(test_agent)
+        assert "error" not in root_span or root_span["error"] == 0
+
+    @missing_feature(context.library == "golang", reason="Not implemented")
+    @missing_feature(context.library == "php", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "ruby", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
```
Comment (mabdinur) [inline tests/parametric/test_otel_span_methods.py:870]:
```suggestion
    @missing_feature(context.library < "python@2.9.0", reason="Not implemented")
```

File: tests/parametric/test_otel_span_methods.py [hunk 3]
```diff
@@ -789,6 +790,128 @@ def test_otel_span_extended_reserved_attributes_overrides_analytics_event(
...
+        assert event3["attributes"].get("int_array")[0] == 3
+        assert event3["attributes"].get("int_array")[1] == 4
+        assert event3["attributes"].get("string_array")[0] == "5"
+        assert event3["attributes"].get("string_array")[1] == "6"
+
+    @missing_feature(context.library == "golang", reason="Not implemented")
+    @missing_feature(context.library == "php", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "ruby", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
```
Comment (mabdinur) [inline tests/parametric/test_otel_span_methods.py:850]:
```suggestion
    @missing_feature(context.library == "nodejs", reason="Not implemented")
    @missing_feature(context.library < "python@2.9.0", reason="Not implemented")
```

**Review by mabdinur — APPROVED** [review]
The Python change that adds support for recording exception as events was merged (and will hopefully be released in v2.9.0 🤞 )

**Pull request merged**